opta/core/kubernetes.py

```diff
@@ -66,24 +66,63 @@ def get_required_path_executables(cloud: str) -> FrozenSet[str]:
     return frozenset({"kubectl"}) | exec_map.get(cloud, set())
 
 
-def configure_kubectl(layer: "Layer") -> None:
-    """Configure the kubectl CLI tool for the given layer"""
+def set_kube_config(layer: "Layer") -> None:
+    """Create a kubeconfig file to connect to a kubernetes cluster specified in a given layer"""
     # Make sure the user has the prerequisite CLI tools installed
     # kubectl may not *technically* be required for this opta command to run, but require
     # it anyways since user must install it to access the cluster.
     ensure_installed("kubectl")
     makedirs(GENERATED_KUBE_CONFIG_DIR, exist_ok=True)
     if layer.cloud == "aws":
-        _aws_configure_kubectl(layer)
+        _aws_set_kube_config(layer)
     elif layer.cloud == "google":
-        _gcp_configure_kubectl(layer)
+        _gcp_set_kube_config(layer)
     elif layer.cloud == "azurerm":
-        _azure_configure_kubectl(layer)
+        _azure_set_kube_config(layer)
     elif layer.cloud == "local":
-        _local_configure_kubectl(layer)
+        _local_set_kube_config(layer)
 
```

Comment:
should we throw an error if not implemented (else case)?

Reply:
good point